Frontend/src/components/Input/DateInput.svelte

```diff
@@ -28,29 +29,34 @@
     ["November", "Nov"],
     ["Dezember", "Dez"],
   ];
+  const TIMEZONE_OFFSET_MS = new Date().getTimezoneOffset() * 60000;
 
   function inTwoMonths() {
     const date = new Date();
     date.setMonth(date.getMonth() + 2);
     return date;
   }
 
-  function clear() {
-    isNone = true;
-    dateChosen = false;
-    selected = new Date();
+  export let calculate_from;
```

Comment:
Do we really need this prop? Doesn't this always equal timeMillis?

Reply:
`calculate_from`? In this case it gives the basis on which +x days is calculated from, e.g. `rented_on`. If calculate_from is 2.3.21, but today is `3.3.21`, `+7` would give `9.3.21`, if calculate_from is None, it would give `10.3.21`

Reply:
by the way: is there no way to work with only dates and not with datetimes? just to prevent any timezone problems?

Reply:
> `calculate_from`? Yes! In this case it gives the basis on which +x days is calculated from, that makes it possible to e.g. calculate it based on `rented_on` and not on `today`. If `calculate_from` is `2.3.21` (=rented_on), but today is `3.3.21`, `+7` would give `9.3.21` (rented_on+7), if calculate_from is None, it would give `10.3.21` (today+7)

Ok, but shouldn't the basis always be rented_on / to_return_on / ... which equals timeMillis? In which cases would we need calculate_from to be today?

Reply:
> by the way: is there no way to work with only dates and not with datetimes? just to prevent any timezone problems?

I know this isn't optimal, but I haven't found a better way yet. JS Date Objects always have a time.

Reply:
> Ok, but shouldn't the basis always be rented_on / to_return_on / ... which equals timeMillis? In which cases would we need calculate_from to be today?

but it's the `timeMillis` of a different element! Or am I missing something? We are not calculating it on the current datepicker elements `timeMillis`, but on another one - which we can;t access directly?

Reply:
> > Ok, but shouldn't the basis always be rented_on / to_return_on / ... which equals timeMillis? In which cases would we need calculate_from to be today?
> 
> but it's the `timeMillis` of a different element! Or am I missing something? We are not calculating it on the current datepicker elements `timeMillis`, but on another one - which we can;t access directly?

'timeMillis' is a prop exported by our DateInput component. It is always bound to rented_on / to_return_on / .... So if either timeMillis or to_return_on changes the other one will be updated.

So I think in the DateInput component we could just update 'timeMillis' when a button is clicked. We could remove "calculate_from", since it would always equal the selected date (timeMillis). When clicking a button we always want to add to the selected date, not to the current day, correct?

Reply:
Even though I suspect that I misunderstand something how Svelte works....

But here we have several date controls:
Ausgeliehen am -> doc.rented_on -> Datepicker1(timeMillis_1)
Zurückerwartet -> doc.to_return_on -> Datepicker2(timeMillis_2)

Now, if I want the `+7d` button inside Zurückerwartet to add 7 days to timeMillis_1 (as we discussed), the only way to know about timeMillis_1 (or doc._to_return_on respectively) is by passing it inside as an argument. As Datepicker2 is agnostic about its parent, we need to pass it as a prop, right? Of course if we want to add +7 just to the timeMillis_2, we dont need it. But for generalizability of the control this would be necessary, or?

Reply:
Hm, yes when pressing +7 multiple times it would add a week every time. But I don't think that would be so bad? You could still reset it by manually selecting another date. If we want to prevent this we could do that inside the DateInput component. Each time a button is pressed we could store the days we added and adapt the number added for the next button press accordingly. This would also work with an additional calculate_from prop, but I think it would be better to keep the number of props small.

At the moment rented_on / to_return_on / ... are passed to the DateInput component as the timeMillis Prop. For example rented_on would equal 'timeMillis' in one instance of a DateInput while to_return_on would equal 'timeMillis' in another instance of a DateInput.

Reply:
Okay, so that makes sense. I would vote for keeping it static, as it can easily happen that it will be pressed accidentially twice, and it becomes more confusing when we want to change from e.g. two week rental to one week rental. In our case what we want is an easy-to-reach control for 1,2 or 3 week rentals, starting from the rented_on, or? And I think that is the most intuitive way of using that button in this case?

Reply:
Yes, keeping it static e.g. for 1,2,3 weeks would be nice. I just meant to do that without the additional calculate_from prop.

Reply:
Ok, but then we can only calculate it based on either the current value of timeMillis (which is already +7?) or from today?

Reply:
> Ok, but then we can only calculate it based on either the current value of timeMillis (which is already +7?) or from today? In our use case that would mean the value of `to_return_on` is initialized already with +7, so when pressing +7 it would add to +14 in total, I don't think that is what we want, right?

I thought that was what we wanted? Because for to_return_on +7 is the default (+0 would not make sense). But I think then the +14 or more buttons would not make sense. 

Why don't we only display a +7 button and allow multiple button presses? So without clicking the button "to_return_on" would be in one week, when pressing the button once in two weeks and when pressing the button twice in three weeks? We could also add a -7 button.

Another possibility might be to label the buttons with dates. So if today is 04.01. we could display 3 buttons for dates "11.01", "18.01" and "25.01". Then it might be more clear what each button does.

Reply:
Mhh, I'm strongly against multiple presses, it is so easy to accidentially press too much and then having to open the calendar and reverting.

Displaying the target dates might be an idea that I could agree with. Then we could also just have one prop called "quick_date" and pass a list of datestamps. However, I actually still prefer the current versioun ^^ (sorry for the lengthy discussion, hope you are not too annoyed by it)

Reply:
Hm ok, I think I could agree with one of these two versions:

- Instead of „+7“ / „+ 14“ we could label the buttons „in 1 Woche“ / „in 2 Wochen“. I think then it would be clear that it is added to today and not to the selected value. I think this is what you meant, correct?
- I also like the the date version. I think if we show three date buttons users will learn fast that it‘s always +1 / +2 / +3 weeks and won‘t think about it every time.

With both versions, the first button for to_return_on wouldn‘t do anything because that‘s already the default value correct?

Reply:
It would do something, in case you pressed +2W, and want to return to +1W. But I have to admit that I don't know how often that will be used .

I thought more of the generalization aspect of the control, in case we want to use it elsewhere.

But let's go with version 1 then :) it actually is already implemented with `quick_weeks `, we just need to change the wording.

Else I thought to have just one prop `quickset = { %N_DAYS% : %DESCRIPTION% }` - what do you think of this? then each button could be customizable.